tbdex/rfq/rfq_test.go

```diff
@@ -81,7 +81,117 @@ func TestRFQ_Sign(t *testing.T) {
 
 func TestRFQ_UnmarshalJSON(t *testing.T) {
 	pfiDID, _ := didjwk.Create()
+	walletDID, _ := didjwk.Create()
+	offeringID, _ := typeid.WithPrefix(offering.Kind)
+
+	r, _ := rfq.Create(
+		walletDID.URI,
+		pfiDID.URI,
+		offeringID.String(),
+		rfq.Payin("100", "STORED_BALANCE"),
+		rfq.Payout("BANK_ACCOUNT"),
+	)
+
+	_ = r.Sign(walletDID)
+
+	bytes, err := json.Marshal(r)
+	assert.NoError(t, err)
+
+	var rfq rfq.RFQ
+	err = rfq.UnmarshalJSON(bytes)
+	assert.NoError(t, err)
+}
+
+func TestRFQ_Unmarshal_Invalid(t *testing.T) {
+	input := []byte(`{"doo": "doo"}`)
+
+	var rfq rfq.RFQ
+	err := rfq.UnmarshalJSON(input)
+	assert.Error(t, err)
+}
+
+func TestRFQ_Verify_NoPrivateDataStrict(t *testing.T) {
```

Comment:
Confused by this test name. "no private data" sounds like the private data has been detached, which should cause `rfq.Verify()` to fail. is the private data being detached somehow?

Reply:
There's no private data in the RFQ at all.

Reply:
It's checking for `for strict = true, if all the hashes are null, then is private data allowed to be null`